CMSIS/RTOS2/FreeRTOS/Source/cmsis_os2.c

```diff
@@ -2521,6 +2518,7 @@ osMemoryPoolId_t osMemoryPoolNew (uint32_t block_count, uint32_t block_size, con
       /* Attributes not provided, allocate memory on heap */
       mem_cb = 0;
       mem_mp = 0;
+      name = NULL;
```

Comment:
Please move this back to its original location (line 2485).
I know `name` is assigned two times when attributes are provided but the whole module tends to initialize variables somewhere at the top and modifies them later after some decision making. It just makes things simple and consistent - and we can survive extra CPU instructions when building without optimization.

Reply:
Done